californium-core/src/main/java/org/eclipse/californium/core/network/EndpointManager.java

```diff
@@ -358,9 +258,12 @@ public void deliverRequest(Exchange exchange) {
 
 		@Override
 		public void deliverResponse(Exchange exchange, Response response) {
-			if (exchange == null) throw new NullPointerException();
-			if (exchange.getRequest() == null) throw new NullPointerException();
-			if (response == null) throw new NullPointerException();
+			if (exchange == null)
+				throw new NullPointerException("no CoAP exchange!");
```

Comment:
Does it make sense to throw an exception here? It would probably be better to log the fact that there is no exchange and then ignore the response.

WDYT?